src/KafkaFlow/Consumers/KafkaConsumerFlowManager.cs

```diff
@@ -0,0 +1,116 @@
+namespace KafkaFlow.Consumers
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Confluent.Kafka;
+
+    internal class KafkaConsumerFlowManager : IKafkaConsumerFlowManager
+    {
+        private readonly IConsumer<byte[], byte[]> consumer;
+        private readonly CancellationToken stopCancellationToken;
+        private readonly ILogHandler logHandler;
+
+        private CancellationTokenSource heartbeatTokenSource;
+        private Task heartbeatTask;
+
+        private readonly List<TopicPartition> pausedPartitions = new List<TopicPartition>();
+
+        public KafkaConsumerFlowManager(
+            IConsumer<byte[], byte[]> consumer,
+            CancellationToken stopCancellationToken,
+            ILogHandler logHandler)
+        {
+            this.consumer = consumer;
+            this.stopCancellationToken = stopCancellationToken;
+            this.logHandler = logHandler;
+        }
+
+        public void Pause(IReadOnlyCollection<TopicPartition> topicPartitions)
+        {
+            lock (this.pausedPartitions)
+            {
+                topicPartitions = topicPartitions.Except(this.pausedPartitions).ToList();
+
+                if (!topicPartitions.Any())
+                {
+                    return;
+                }
+
+                this.consumer.Pause(topicPartitions);
+                this.pausedPartitions.AddRange(topicPartitions);
+
+                if (this.HasRunningPartitions())
+                {
+                    return;
+                }
+
+                this.heartbeatTokenSource = CancellationTokenSource.CreateLinkedTokenSource(this.stopCancellationToken);
+
+                this.heartbeatTask = Task.Run(
+                    () =>
+                    {
+                        try
+                        {
+                            while (!this.heartbeatTokenSource.IsCancellationRequested)
+                            {
+                                var result = this.consumer.Consume(1000);
```

Comment:
Why this timeout? 
I guess it should block undefinitely, dont you think?

Reply:
I mean, undefinitely = until the task IS cancelled by `Resume`.

Reply:
In this case, the Consume() call will wait until this time expires. If we put a smaller value we will put some unnecessary load on  Kafka brokers, if we put a greater value we will increase the time that Resume() will block the execution. I put the `1000` in a `const int` to name it